protocol/test/foundry/PipelineConvert.t.sol

```diff
@@ -0,0 +1,1461 @@
+
+        passGermination();
+
+        // do the convert
+
+        // Create arrays for stem and amount. Tried just passing in [stem] and it's like nope.
+        int96[] memory stems = new int96[](1);
+        stems[0] = stem;
+
+        AdvancedFarmCall[] memory farmCalls = new AdvancedFarmCall[](1);
+        AdvancedFarmCall[] memory beanToLPFarmCalls = createBeanToLPFarmCalls(amount);
+        farmCalls[0] = beanToLPFarmCalls[0]; // Assign the first element of the returned array
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = amount;
+
+        // get well amount out if we deposit 200 beans
+        uint256 wellAmountOut = getWellAmountOutForAddingBeans(amount);
+
+        // verify convert
+        vm.expectEmit(true, false, false, true);
+        emit Convert(users[1], C.BEAN, C.BEAN_ETH_WELL, amount, wellAmountOut);
+
+        // vm.expectEmit(true, false, false, true);
+        // emit RemoveDeposits(users[1], C.BEAN, stems, amounts, amount, amounts);
+
+        // get bdv of this well amount out for later comparison
+        // FIXME: the well amount out reverts because pumps returning zero, maybe need a helper to update pumps?
+        // uint256 bdvOfThisWellAmountOut = bs.bdv(C.BEAN_ETH_WELL, wellAmountOut);
+
+        // vm.expectEmit(true, false, false, true);
+        // emit AddDeposit(users[1], C.BEAN_ETH_WELL, stem, wellAmountOut, amount);
+
+        
+        vm.resumeGasMetering();
+        vm.prank(users[1]); // do this as user 1
+        convert.pipelineConvert(
+            C.BEAN, // input token
+            stems,  // stems
+            amounts,  // amount
+            C.BEAN_ETH_WELL, // token out
+            farmCalls // farmData
+        );
+
+    }
+
+    function testBasicConvertLPToBean(uint256 amount) public {
+        vm.pauseGasMetering();
+
+        // well is initalized with 10000 beans. cap add liquidity 
+        // to reasonable amounts. 
+        amount = bound(amount, 1e6, 10000e6);
+
+        (int96 stem, uint256 lpAmountOut) = depositLPAndPassGermination(amount);
+
+
+        // Create arrays for stem and amount. Tried just passing in [stem] and it's like nope.
+        int96[] memory stems = new int96[](1);
+        stems[0] = stem;
+
+        AdvancedFarmCall[] memory farmCalls = new AdvancedFarmCall[](1);
+        AdvancedFarmCall[] memory beanToLPFarmCalls = createLPToBeanFarmCalls(lpAmountOut);
+        farmCalls[0] = beanToLPFarmCalls[0]; // Assign the first element of the returned array
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = lpAmountOut;
+        
+        vm.resumeGasMetering();
+        vm.prank(users[1]); // do this as user 1
+        convert.pipelineConvert(
+            C.BEAN_ETH_WELL, // input token
+            stems,  // stems
+            amounts,  // amount
+            C.BEAN, // token out
+            farmCalls // farmData
+        );
+    }
+
+
+
+    function testBasicConvertLPToLP(uint256 amount) public {
```

Comment:
this test should be created such that it performs both a beanETH <> beanwstETH and beanwStETH <> beanETH convert. this can be done with some fuzz randomizer.

Reply:
Done, and updates stalk penalty calc to view function, with a different apply function that updates convert capacities upon actual convert. 
Here: https://github.com/BeanstalkFarms/Beanstalk/pull/840/commits/2d79e3f150630c53bc661e4e144679f916bb36c9 and here: https://github.com/BeanstalkFarms/Beanstalk/pull/840/commits/70c6088af1a74e95403d1f0228c1499713262854